packages/toc/src/tocitem.tsx

```diff
@@ -80,8 +79,8 @@ export class TableOfContentsItem extends React.PureComponent<
             {heading.text}
           </span>
         </div>
-        {children && !heading.collapsed && <ol>{children}</ol>}
-      </li>
+        {children && !heading.collapsed && <>{children}</>}
```

Comment:
As the collapse/expand state is handle for us by the component, we can simply write:

```suggestion
        {children}
```